xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -153,6 +153,82 @@ is_brick_mx_enabled(void)
     return ret ? _gf_false : enabled;
 }
 
+static gf_boolean_t
+gd_has_local_address(glusterd_conf_t *priv, const char *hostname)
+{
+    glusterd_hostname_t *hostname_obj = NULL;
+
+    cds_list_for_each_entry(hostname_obj, &priv->hostnames, hostname_list)
+    {
+        if (strcmp(hostname_obj->hostname, hostname) == 0) {
+            return _gf_true;
+        }
+    }
+
+    return _gf_false;
+}
+
+static int
+glusterd_hostname_new(xlator_t *this, const char *hostname,
+                      glusterd_hostname_t **name)
+{
+    glusterd_hostname_t *hostname_obj = NULL;
+    int32_t ret = -1;
+
+    GF_ASSERT(hostname);
+    GF_ASSERT(name);
+
+    hostname_obj = GF_MALLOC(sizeof(*hostname_obj), gf_gld_mt_hostname_t);
+
+    if (!hostname_obj) {
+        gf_smsg(this->name, GF_LOG_ERROR, errno, GD_MSG_NO_MEMORY, NULL);
+        goto out;
+    }
+
+    hostname_obj->hostname = gf_strdup(hostname);
+    CDS_INIT_LIST_HEAD(&hostname_obj->hostname_list);
+
+    *name = hostname_obj;
+    ret = 0;
+
+out:
+    gf_msg_debug("glusterd", 0, "Returning %d", ret);
+    return ret;
+}
+
+gf_boolean_t
+glusterd_gf_is_local_addr(char *hostname)
+{
+    xlator_t *this = NULL;
+    glusterd_conf_t *priv = NULL;
+    glusterd_hostname_t *hostname_obj = NULL;
+    gf_boolean_t found = _gf_false;
+    int ret = 1;
+
+    this = THIS;
+    priv = this->private;
+
+    if (gd_has_local_address(priv, hostname)) {
+        found = _gf_true;
+        goto out;
+    }
+
+    if (gf_is_local_addr(hostname)) {
+        ret = glusterd_hostname_new(this, hostname, &hostname_obj);
+        if (ret) {
+             gf_smsg(this->name, GF_LOG_ERROR, ENOMEM, GD_MSG_NO_MEMORY,
+                     NULL);
+            goto out;
+        }
+        found = _gf_true;
+        cds_list_add_tail_rcu(&hostname_obj->hostname_list, &priv->hostnames);
```

Comment:
Why do you use an rcu-based list ? if the list will not be used concurrently, a simple `struct list_head` is enough. But if the list can be used concurrently, list modifications need to be protected by a mutex.

Reply:
I see that @mohit84 addressed this concern now. Merging the PR as it helps to fix a test case!